temoa/utilities/graph_utils.py

```
@@ -0,0 +1,238 @@
+        node_entry: dict[str, Any] = {'id': node_id_str}
+        if 'label' not in attrs:
+            node_entry['label'] = node_id_str
+        for key, value in attrs.items():
+            if isinstance(value, (list, dict)):
+                try:
+                    json.dumps(value)
+                    node_entry[key] = value
+                except (TypeError, OverflowError):
+                    if verbosity >= 2:
+                        logger.debug(
+                            f"Node {node_id_str} attr '{key}' not JSON serializable, converting to string."
```

Ruff (0.14.1)

Logging statement uses f-string

(G004)